### PR TITLE
DOCSP-9595: Update Multiple usage example

### PR DESCRIPTION
**Pull request by ccho-mongodb**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-9595

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/6a98491/java/docsworker-xlarge/DOCSP-9595-update-multiple/usage-examples/updateMany

### CODE REVIEW

File: source/includes/usage-examples/code-snippets/UpdateMany.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// ignored first line
```
Comment (nathan-contino-mongo) [inline source/includes/usage-examples/code-snippets/UpdateMany.java:1]: apparently this ignored first line behaviour is going to disappear soon, I can't find the platform ticket for it though.

File: source/usage-examples/updateMany.txt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+Update a Multiple Documents
+===========================
+
+.. default-domain:: mongodb
+
+You can update multiple documents using the ``updateMany()`` method on
+a ``MongoCollection`` object. The method accepts a **filter** that matches the
+document you want to update and an **update** statement that instructs the
+driver to change the matching document. The ``updateMany()`` method only
+updates all the documents in the collection that match the filter.
```
Comment (nathan-contino-mongo) [inline source/usage-examples/updateMany.txt:11]: personally I feel that the "only...all" construction here is a little hard to parse. Do you think we could remove "only", or is there a good reason for it to remain here?

File: source/usage-examples/updateMany.txt [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+To perform an update with the ``updateMany()`` method, you must pass
```
Comment (nathan-contino-mongo) [inline source/usage-examples/updateMany.txt:13]:
```suggestion
To perform an update with the ``updateMany()`` method, you must provide
```
To me, "provide" implies that you should construct these yourself, while "pass" implies that these parameters might already exist and you just have to pass them along. But I admit that it's entirely subjective.

File: source/usage-examples/updateMany.txt [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+a query filter and an update document. The query filter specifies the criteria
+for which document to perform the update on and the update document provides
+instructions on what changes to make to them.
```
Comment (nathan-contino-mongo) [inline source/usage-examples/updateMany.txt:16]: This sentence is a little bit of a mouthful for me, would it make sense to break this up into two sentences, each explaining a single parameter? "which document to perform the update on" also implies that this method only updates a single document -- can we pluralize that?

File: source/usage-examples/updateMany.txt [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+You can optionally pass an instance of ``UpdateOptions`` to the method in
+order to specify the behavior. For example, if you set the ``upsert`` field of
```
Comment (nathan-contino-mongo) [inline source/usage-examples/updateMany.txt:19]:
```suggestion
order to adjust behavior. For example, if you set the ``upsert`` field of
```
I'm not sure if this sounds any better, but I definitely think we should at least ditch the "the" before behavior.

File: source/usage-examples/updateMany.txt [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+Upon successful execution, the ``updateMany()`` method returns an instance
+of ``UpdateResult``. You can retrieve information such as the number of
+documents modified calling the ``getModifiedCount()`` method. If you
+specified ``upsert(true)`` in the ``UpdateOptions`` object and the
+operation results in an insert, you can retrieve the ``_id`` field of the
+new document by calling the ``getUpsertedId()`` method on the
+``UpdateResult``.
+
+If your update operation fails, the driver raises an exception and none of
+the documents matching the filter are updated. For example, if you try to set
```
Comment (nathan-contino-mongo) [inline source/usage-examples/updateMany.txt:33]: "none of the documents matching the filter are update"
Is this strictly true? For instance, if you update half the documents, then violate a unique field constraint for a single document halfway through the update, wouldn't some of the documents (the first half) stay updated?

It might be a good idea to avoid mentioning the side-effects of failed operations entirely since that's likely highly dependent on configuration and when the error occurs. Though I'll grant that in the case you provided, none of the documents would update.

File: source/usage-examples/updateMany.txt [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+- Add ``Frequently Discussed`` to the array of ``genres`` only if it does not
+  already exist
+- Set the value of ``lastUpdated`` to the current time.
+
+We use the ``Updates`` builder, a factory class that contains static
+helper methods to construct the update document. While you can pass an update
+document instead of using the builder, the builder provides type checking and
+simplified syntax. Read our
+:doc:`guide on Updates </fundamentals/builders/updates>` in the Builders
+section for more information.
```
Comment (nathan-contino-mongo) [inline source/usage-examples/updateMany.txt:70]: Love this little bit of background on the ``Updates`` builder, this is a great way to pull slightly more advanced concepts into the usage examples!

File: source/usage-examples/updateMany.txt (hunk 6 above)
Comment (ccho-mongodb) [reply source/usage-examples/updateMany.txt:33]: As far as I've been able to test it, yes. This might be different from "bulkWrite" operations. Will double check with Jeff.

File: source/includes/usage-examples/code-snippets/UpdateMany.java (hunk 1 above)
Comment (ccho-mongodb) [reply source/includes/usage-examples/code-snippets/UpdateMany.java:1]: Release for this feature is tonight, so I'll create a separate PR to address all the issues here and other driver repos.

File: source/usage-examples/updateMany.txt (hunk 2 above)
Comment (ccho-mongodb) [reply source/usage-examples/updateMany.txt:11]: Whoops, good catch. Will fix.

File: source/usage-examples/updateMany.txt (hunk 3 above)
Comment (ccho-mongodb) [reply source/usage-examples/updateMany.txt:13]: They imply the same thing to me and both require qualifiers to discern whether it's a pre-existing object or if you're declaring them inline. In my own experience, I have encountered "pass parameters" more often than "provide parameters" in documentation and other forms of communication.

File: source/usage-examples/updateMany.txt (hunk 5 above)
Comment (ccho-mongodb) [reply source/usage-examples/updateMany.txt:19]: Will address.

File: source/usage-examples/updateMany.txt (hunk 4 above)
Comment (ccho-mongodb) [reply source/usage-examples/updateMany.txt:16]: I think I accidentally left "document" in there, so it doesn't belong there. I'll update to condense the sentence.

File: source/usage-examples/updateMany.txt (hunk 3 above)
Comment (nathan-contino-mongo) [reply source/usage-examples/updateMany.txt:13]: Seems reasonable, more common = clearer for our readers.

File: source/includes/usage-examples/code-snippets/UpdateMany.java (hunk 1 above)
Comment (nathan-contino-mongo) [reply source/includes/usage-examples/code-snippets/UpdateMany.java:1]: nice, thanks for staying on top of that!
